hayeon/programmers-1/로또의-최고-순위와-최저-순위.js

```diff
@@ -0,0 +1,32 @@
+function solution(lottos, win_nums) {
+  const answer = [];
+  let zeroCount = 0;
+  let matchCount = 0;
+
+  for (let num of lottos) {
+    if (num === 0) {
+      zeroCount++;
+    }
+    if (win_nums.includes(num)) {
+      matchCount++;
+    }
+  }
+
+  const maxRank =
+    zeroCount === 0 && matchCount === 0 ? 6 : 7 - (zeroCount + matchCount);
+
+  const minRank = matchCount === 0 ? 6 : 7 - matchCount;
+
+  answer.push(maxRank, minRank);
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  let zeroCount = 0;
  let matchCount = 0;

  for (let num of lottos) {
    if (num === 0) {
      zeroCount++;
    }
      
    if (win_nums.includes(num)) {
      matchCount++;
    }
  }

  const maxRank = zeroCount + matchCount;

  return [rank[maxRank], rank[matchCount]];
}
```
- 풀이 잘봤습니다 🙇🏻‍♂️
- 랭크를 매핑해주는 배열 (rank)를 활용한다면 좀 더 깔끔한 풀이가 가능합니다 :) 
- 아래는 filter를 활용한 풀이인데 참고해주시면 좋을 것 같습니다 ! 
```js
function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  const winCount = lottos.filter((lotto) => win_nums.includes(lotto)).length;
  const zeroCount = lottos.filter((lotto) => !lotto).length;
  const bestCount = winCount + zeroCount;

  return [rank[bestCount], rank[winCount]];
}
```